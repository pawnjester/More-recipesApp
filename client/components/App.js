import React from 'react';
import style from '../styles/index.scss';

export default () => {
  return (
    <h1>Hello World</h1>
  );
}