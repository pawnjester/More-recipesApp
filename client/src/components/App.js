import React from 'react';
import '../styles/index.scss';
import NavigationBar from './NavigationBar';
import Main from './Main';
import FlashMessagesList from './flash/FlashMessagesList';

class App extends React.Component {
  render() {
    return (
      <div className= 'container'>
        <NavigationBar />
        <FlashMessagesList />
        <Main />
      </div>
    );
  }
}

export default App;