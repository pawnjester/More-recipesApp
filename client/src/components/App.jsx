import React from 'react';
import Routes from './Routes';

export const App = () => (
  <div className="wrap">
    <Routes />
  </div>
);


export default App;
