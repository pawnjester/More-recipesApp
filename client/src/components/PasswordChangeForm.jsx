import React, { Component } from 'react';
import { connect } from 'react-redux';
import toastr from 'toastr';
import { Redirect } from 'react-router-dom';
import resetPassword from '../actions/resetPassword';
import NavigationBarComponent from './NavigationBar';
import validateInput from './validations/confirmNewPassword';

/**
 *
 * @class PasswordChangeForm
 *
 * @extends {Component}
 */
class PasswordChangeForm extends Component {
  /**
   * @description Creates an instance of PasswordChangeForm.
   *
   * @param {any} props
   *
   * @memberof PasswordChangeForm
   */
  constructor(props) {
    super(props);
    this.state = {
      password: '',
      passwordConfirmation: '',
      errors: {},
      redirect: false,
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  /**
   * @description set the input value to state
   *
   * @param {object} event
   *
   * @memberof PasswordChangeForm
   *
   * @returns {void}
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
 * @description onSubmit function
 *
 * @param {object} event
 *
 * @memberof PasswordChangeForm
 *
 * @returns {void}
 */
  onSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      const { token } = this.props.match.params;
      const { password, passwordConfirmation } = this.state;

      this.props.resetPassword(token, password, passwordConfirmation).then(() => {
        this.setState({ redirect: true });
        toastr.success('Password successfully changed, Log in now');
      }, )
        .catch((error) => {
          this.setState({ errors: error.response.data });
          toastr.error(`${this.state.errors.error}`);
        });
    }
  }

  /**
   * @description inValid function
   *
   * @memberof PasswordChangeForm
   *
   * @returns {void}
   */
  isValid() {
    const { errors, isValid } = validateInput(this.state);

    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
 * @description renders the jsx element
 *
 *
 * @memberof PasswordChangeForm
 *
 * @returns {void}
 */
  render() {
    const { redirect, errors } = this.state;
    if (redirect) {
      return <Redirect to="/newLogin" />;
    }
    return (
      <div>
        <NavigationBarComponent />
        <div className="container">
          <div className="row mt-5 ml-5">
            <div
              className="card "
              style={{ width: 700, marginLeft: 130, borderColor: '#ec7026' }}
            >
              <div className="card-body">
                <h4 className=" text-center mr-5 mb-5 ">Change your password</h4>
                <form className="form-group">
                  <div className="form-group">
                    <div>
                      <input
                        type="password"
                        placeholder="password"
                        name="password"
                        onChange={this.onChange}
                        value={this.state.password}
                        className="p-1 out"
                        style={{
                          width: 400, borderRadius: `${10}px`, borderColor: '#ff4500', marginLeft: 130
                        }}
                      />
                    </div>
                    {(errors.error || errors.passwordConfirmation)
                      && <small style={{ marginLeft: 250 }}>
                        {(errors.error || errors.passwordConfirmation)}
                      </small>}
                  </div>
                  <div className="form-group">
                    <div>
                      <input
                        type="password"
                        placeholder="Confirm password"
                        name="passwordConfirmation"
                        onChange={this.onChange}
                        value={this.state.passwordConfirmation}
                        className="p-1 out"
                        style={{
                          width: 400,
                          borderRadius: `${15}px`,
                          borderColor: '#ff4500',
                          marginLeft: 130
                        }}
                      />
                    </div>
                    {(errors.error || errors.passwordConfirmation)
                      && <small style={{ marginLeft: 250 }}>
                        {(errors.error || errors.passwordConfirmation)}
                         </small>}
                  </div>
                  <button
                    onClick={this.onSubmit}
                    type="button"
                    className="btn btn-outline-primary pr-3 pl-3 password-color"
                    style={{
                      borderRadius: 100,
                      backgroundColor: '#ec7026',
                      marginLeft: 250,
                      color: 'black'
                    }}
                  > Change Password
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}


export default connect(null, { resetPassword })(PasswordChangeForm);
