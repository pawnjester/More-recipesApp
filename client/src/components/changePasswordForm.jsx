import React, { Component } from 'react';
import { connect } from 'react-redux';
import toastr from 'toastr';
import NavigationBar from './NavigationBar';
import changePassword from '../actions/changePassword';
import '../styles/passwordchange.scss';

/**
 *
 *
 * @class ChangePasswordForm
 *
 * @extends {Component}
 */

class ChangePasswordForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      oldPassword: '',
      password: ''
    };
  };

  onNameChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  onSubmit = (event) => {
    event.preventDefault();
    this.props.changePassword(this.state).then((res) => {
      toastr.success('Password successfully changed');
      this.setState({
        oldPassword: '',
        password: ''
      });
    })
      .catch(error => toastr.error(error.response.data.error));
  }

  render() {
    return (
      <div >
        <NavigationBar />
        <div>
          <form className="mx-auto app-login-form card-login">
            <h4 className="black text-center mt-5">Change Your Password</h4>
            <section className="pt-4 input-group">
              <span className="input-group-addon">
                <i className="fa fa-lock black" aria-hidden="true"></i>
              </span>
              <input
                name="oldPassword"
                className="form-control"
                type="password"
                value={this.state.oldPassword}
                onChange={this.onNameChange}
                placeholder="Enter your old password"
              />
            </section>
            <section className="pt-4 pb-3 input-group">
              <span className="input-group-addon">
                <i className="fa fa-lock black" aria-hidden="true"></i>
              </span>
              <input
                name="password"
                className="form-control"
                type="password"
                value={this.state.password}
                onChange={this.onNameChange}
                placeholder="Enter a new Password"
              />
            </section>
            <div className="pt-2">
              <button
                className="btn btn-block text-capitalize password-color"
                onClick={this.onSubmit}
              >
                change Password
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

export default connect(null, { changePassword })(ChangePasswordForm);
