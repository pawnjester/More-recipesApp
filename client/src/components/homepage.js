import React from 'react';
import '../styles/home.scss';
import Signupform from './signup/signupform';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignupRequest } from '../actions/signupActions';
import { Button } from 'react-bootstrap';



class Home extends React.Component {
  render() {
    const {userSignupRequest} = this.props;

    return (
    <header id ="home-section">
      <div className= "dark-overlay">
      <div className= "home-inner">
        <div className ="container">
          <div className= "row">
            <div className="col-lg-8 d-none d-lg-block">
              <h1 className= "display-4" style = {{color: 'fff'}}>Let the  world know about your <strong>Recipe</strong> in three easy steps:</h1>
              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Create an account
                </div>
                
              </div>

              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Add A Recipe 
                </div>
                
              </div>

              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Relax, and let the world view your wonderful recipe
                </div>
                
              </div>
            </div>
            <div className="col-lg-4">
              <div className= "card bg-danger text-center card-form">
                <div className="card-body">
                  <h3>Sign Up Today</h3>
                  <p>Please fill this form to register</p>
                    <Signupform userSignupRequest = {userSignupRequest} />
                </div>
              </div>
            </div>
          </div>
          
        </div>
        
      </div>
      </div>
      
    </header >
    )
  }
}

Home.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
}

export default connect(null, { userSignupRequest})(Home)