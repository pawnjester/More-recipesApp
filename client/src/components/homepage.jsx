import React from 'react';
import Signupform from './signup/signupform';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignupRequest } from '../actions/signupActions';
import { Button } from 'react-bootstrap';
import NavigationBar from './NavigationBar';
import '../styles/home.scss';
/*eslint-disable */


class Home extends React.Component {
  render() {
    const { userSignupRequest } = this.props;

    return (
      <div>
        <NavigationBar />
        <header id="home-section">
          <div className="dark-overlay">
            <div className="home-inner">
              <div className="container">
                <div className="row">
                  <div className="col-lg-8 d-none d-lg-block">
                <h1 className="display-4" style={{ color: 'fff' }}>Let the  world know about your <strong>Recipe</strong> in three easy steps:</h1>
                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end ">
                  Create an account
                </div>

                </div>

                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end ">
                  Add A Recipe
                </div>

                </div>

                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end">
                  Relax, and let the world view your wonderful recipe
                </div>

                </div>
              </div>
                  <div className="col-lg-4">
                <div className="card bg-danger text-center card-form">
                  <div className="card-body">
                  <h3>Sign Up Today</h3>
                  <p>Please fill this form to register</p>
                  <Signupform userSignupRequest={userSignupRequest} />
                </div>
                </div>
              </div>
                </div>

              </div>

            </div>
          </div>

        </header >
      </div>
    );
  }
}

Home.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
};

export default connect(null, { userSignupRequest })(Home);
