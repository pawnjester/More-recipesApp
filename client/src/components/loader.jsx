import React from 'react';
import { RingLoader } from 'react-spinners';


const Loader = props => (
  <div className="sweet-loading loader-center">
    <RingLoader
      color="#e50000"
      loading={props.loading}
    />
  </div>
);

export default Loader;
