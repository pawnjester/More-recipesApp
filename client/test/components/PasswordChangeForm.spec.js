import React from 'React';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import sinon from 'sinon';
import { provider } from 'react-redux';
import { PasswordChangeForm } from '../../src/components/PasswordChangeForm';
import store from '../../src/store/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';


configure({ adapter: new Adapter() });

const props = {
  resetPassword: jest.fn()
}
function setup() {
  const shallowWrapper = shallow(<PasswordChangeForm {...props} />);
  return {
    shallowWrapper,
  };
}
describe('Test For change Password Form', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('onChange() should', () => {
  it('be called when form input changes', () => {
    const { shallowWrapper } = setup();
    const event = {
      target: { name: 'password', value: 'randomxxcom' }
    };
    const inputForm = shallowWrapper.find('.password');
    inputForm.simulate('change', event);
    expect(shallowWrapper.instance().state.password).toBe('randomxxcom');
  });
});
// describe('onSubmit', () => {
//   it('should not work for invalid email', () => {
//     const wrapper = shallow(<PasswordChangeForm {...props} />)
//     const event = {
//       preventDefault: jest.fn();
//       const form = wrapper.find('.password-color');
//       wrapper.setState({ email: ''});
//       form.simulate('click', event);
//     }
//   })
// })
