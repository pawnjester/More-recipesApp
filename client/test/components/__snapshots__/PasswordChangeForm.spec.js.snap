// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test For change Password Form should render correctly 1`] = `
<div>
  <Connect(NavigationBar) />
  <div
    className="container"
  >
    <div
      className="row mt-5 ml-5"
    >
      <div
        className="card "
        style={
          Object {
            "borderColor": "#ec7026",
            "marginLeft": 130,
            "width": 700,
          }
        }
      >
        <div
          className="card-body"
        >
          <h4
            className=" text-center mr-5 mb-5 "
          >
            Change your password
          </h4>
          <form
            className="form-group"
          >
            <div
              className="form-group"
            >
              <div>
                <input
                  className="p-1 out password"
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  style={
                    Object {
                      "borderColor": "#ff4500",
                      "borderRadius": "10px",
                      "marginLeft": 130,
                      "width": 400,
                    }
                  }
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              className="form-group"
            >
              <div>
                <input
                  className="p-1 out"
                  name="passwordConfirmation"
                  onChange={[Function]}
                  placeholder="Confirm password"
                  style={
                    Object {
                      "borderColor": "#ff4500",
                      "borderRadius": "15px",
                      "marginLeft": 130,
                      "width": 400,
                    }
                  }
                  type="password"
                  value=""
                />
              </div>
            </div>
            <button
              className="btn btn-outline-primary pr-3 pl-3 password-color"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#ec7026",
                  "borderRadius": 100,
                  "color": "black",
                  "marginLeft": 250,
                }
              }
              type="button"
            >
               Change Password
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
